norms/production_application_checklist.md

```diff
@@ -0,0 +1,176 @@
+# Goal of this Documentation
+
+Audience for this document: This is an internal document for Princeton University Library IT Developer teams to use in developing and maintaining applications.
+
+Below is a list of expectations that we have for the systems that we run in production. Together the goal is to make our applications easy to maintain, simple to respond to incidents for, enable experience maintaining applications to be portable between teams, and increase work satisfaction for developers. These are collective requirements \- no individual is expected to implement all of these themselves, but by working together we can implement these for our applications.
+
+Our experience has been that the time it takes to implement these practices is exceeded in the long run by the time and stress we save in doing so. It should be possible for all team members to have low stress, take vacations when they want, and feel comfortable in the environment our department builds.
+
+How we use this document
+
+* As a roadmap when starting new projects  
+* As a guide for improving existing applications
+
+# Checklist
+
+There are some examples of technologies in a few of these requirements, but they are only there for illustrative purposes. It doesn’t particularly matter how these practices are implemented, just that they are.
+
+## Infrastructure
+
+### Infrastructure as Code
+
+#### Reasoning
+
+Defining our infrastructure as code lets us easily reproduce an environment if it needs to be replaced or upgraded. Manual intervention on machines makes them more fragile, so we avoid that with infrastructure as code technology. An example is Ansible. If there is no way to define the infrastructure as code, then the steps to get to the desired state should be documented as a run book.
+
+### One Application Per Server
+
+#### Reasoning
```

Comment:
I'm not sure the "Reasoning" header adds a lot in cases like this where it's the whole section. Maybe skip it unless there is a need to differentiate the reasoning from something else?

Reply:
I just liked the symmetry.

Reply:
That's fine — it's a stylistic thing that's not really important.